adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala

```diff
@@ -85,6 +86,78 @@ sealed trait AlignmentRecordRDD extends AvroReadGroupGenomicRDD[AlignmentRecord,
   }
 
   /**
+   * Convert this set of reads into Coverage records
+   *
+   * @return Returns a CoverageRDD where all reads have been grouped and
+   * counted at all overlapping positions
+   */
+  def toCoverage: CoverageRDD = {
+    val coverage: RDD[(ReferenceRegion, Int)] =
+      rdd.rdd
+        .flatMap(r => {
+          val t: List[Long] = List.range(r.getStart, r.getEnd)
+          t.map(n => (ReferenceRegion(r.getContigName, n, n + 1), 1))
+        }).reduceByKey(_ + _)
+        .cache()
+        .sortByKey()
+        .mapPartitions(i => merge(i)) // merge adjacent regions with the same coverage
```

Comment:
Sorry, forgot to finish writing the thought after the orphan `Also,`. Wherever you have a `cache`, that should be paired with an `unpersist`.